Bottom: 47306cc9a8685a0ee49ade19dd3f615172687732
Top:    3594d6b9c71842b7297ab8e49baf648b80e715dd
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-01-18 14:33:39 +0800

Refresh of tun-rx-batching

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 8fc90c1..485a252f 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -2492,9 +2492,9 @@ static int tun_set_coalesce(struct net_device *dev,
 	struct tun_struct *tun = netdev_priv(dev);
 
 	if (ec->rx_max_coalesced_frames > NAPI_POLL_WEIGHT)
-		return -EINVAL;
-
-	tun->rx_batched = ec->rx_max_coalesced_frames;
+		tun->rx_batched = NAPI_POLL_WEIGHT;
+	else
+		tun->rx_batched = ec->rx_max_coalesced_frames;
 
 	return 0;
 }
